bsp/stm32/libraries/HAL_Drivers/drv_pwm.c

```diff
@@ -291,7 +303,13 @@ static rt_err_t drv_pwm_set(TIM_HandleTypeDef *htim, struct rt_pwm_configuration
     rt_uint32_t period, pulse;
     rt_uint64_t tim_clock, psc;
     /* Converts the channel number to the channel number of Hal library */
-    rt_uint32_t channel = 0x04 * (configuration->channel - 1);
+    rt_uint32_t channel = configuration->channel;
+    if(channel<1 || channel >4)
```

Comment:
这个位置细节, 下面还有类似的 请修改一下：
```suggestion
    if(channel <1 || channel > 4)
```

Reply:
这个已经改了，其他系列有pwm的看了下资料，基本都是支持4路pwm，所以这个算式应该还是通用的，可以合并。

Reply:
好的 感谢提交PR！